adagrams/game.py

```diff
@@ -1,11 +1,157 @@
+from random import choices
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+# Wave 1
+
 def draw_letters():
-    pass
+    ''' Return a hand of 10 randomly selected letters for user.'''
+    letter_bank = {}
+
+    # Randomly select a letter according to its weight in LETTER_POOL
+    while sum(letter_bank.values()) < 10:
+        letter_pool_keys = list(LETTER_POOL.keys())
+        weighted_values = list(LETTER_POOL.values())
+        # random.choices(population, weights, k) --> returns single letter list
+        letter = choices(letter_pool_keys, weighted_values, k=1)
+        letter = letter[0]
+    
+        if letter in letter_bank:
+            # Ensure letter does not occur more frequently than in LETTER_POOL 
+            if letter_bank[letter] >= LETTER_POOL[letter]:
+                pass
+            else:
+                # Increment letter frequency
+                letter_bank[letter] = letter_bank.get(letter, 0) + 1
+        else:
+            # Add letter
+            letter_bank[letter] = 1
```

Comment:
This works, but could be a bit simpler:

```suggestion
        if letter_bank.get(letter):
            if letter_bank[letter] < LETTER_POOL[letter]:
                letter_bank[letter] += 1
        else:
            letter_bank[letter] = 1
```

**However** There's something wrong if it's possible to select a letter more often than it appears in the LETTER_POOL.  Instead I suggest just making a list of all the possible letters and selecting and removing from that list.